server/src/test/java/com/angelomelonas/grpcwebchat/integration/StreamObserverTestHelper.java

```diff
@@ -0,0 +1,84 @@
+package com.angelomelonas.grpcwebchat.integration;
+
+import io.grpc.stub.StreamObserver;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class StreamObserverTestHelper<T> implements StreamObserver<T> {
+
+    private final Object completeLock = new Object();
+    private boolean completed = false;
+    private final List<T> results = new ArrayList<>();
+    private Optional<T> result = Optional.empty();
+    private final Object onNextLock = new Object();
+    private boolean onNext = false;
+
+    @Override
+    public void onNext(T value) {
+        result = Optional.ofNullable(value);
+        results.add(value);
+        notifyOnNext();
+    }
+
+    @Override
+    public void onError(Throwable t) {
+
```

Comment:
you probably want to atleast log the exception ... really helps when it comes to debugging tests

Reply:
Will do.